sql/core/src/test/scala/org/apache/spark/sql/execution/joins/InnerJoinSuite.scala

```diff
@@ -228,6 +230,27 @@ class InnerJoinSuite extends SparkPlanTest with SharedSQLContext {
     )
   )
 
+  testInnerJoin(
```

Comment:
It still can pass without the changes in this PR. What is the purpose of this test case?

Reply:
This test fails in 2.1.1 and versions before https://github.com/apache/spark/commit/6b6dd682e84d3b03d0b15fbd81a0d16729e521d2. I'm not sure how to exercise the code generated by CodegenFallback with that fix, but this test is valid for the 2.1.1 branch.